Sources/FoundationEssentials/Subprocess/Subprocess+Configuration.swift

```diff
@@ -0,0 +1,615 @@
+            return .init(config: .inherit(newValue.wrapToStringOrRawBytes()))
+        }
+
+        public static func custom(_ newValue: [String : String]) -> Self {
+            return .init(config: .custom(newValue.wrapToStringOrRawBytes()))
+        }
+
+        public static func custom(_ newValue: [Data : Data]) -> Self {
+            return .init(config: .custom(newValue.wrapToStringOrRawBytes()))
+        }
+    }
+}
+
+fileprivate extension Dictionary where Key == String, Value == String {
+    func wrapToStringOrRawBytes() -> [Subprocess.StringOrRawBytes : Subprocess.StringOrRawBytes] {
+        var result = Dictionary<
+            Subprocess.StringOrRawBytes,
+            Subprocess.StringOrRawBytes
+        >(minimumCapacity: self.count)
+        for (key, value) in self {
+            result[.string(key)] = .string(value)
+        }
+        return result
+    }
+}
+
+fileprivate extension Dictionary where Key == Data, Value == Data {
+    func wrapToStringOrRawBytes() -> [Subprocess.StringOrRawBytes : Subprocess.StringOrRawBytes] {
+        var result = Dictionary<
+            Subprocess.StringOrRawBytes,
+            Subprocess.StringOrRawBytes
+        >(minimumCapacity: self.count)
+        for (key, value) in self {
+            result[.rawBytes(key.toArray())] = .rawBytes(value.toArray())
+        }
+        return result
+    }
+}
+
+fileprivate extension Data {
+    func toArray<T>() -> [T] {
+        return self.withUnsafeBytes { ptr in
+            return Array(ptr.bindMemory(to: T.self))
+        }
+    }
+}
+
+// MARK: - ProcessIdentifier
+extension Subprocess {
+    public struct ProcessIdentifier: Sendable, Hashable {
+        let value: pid_t
+
+        internal init(value: pid_t) {
+            self.value = value
+        }
+    }
+}
+
+// MARK: - TerminationStatus
+extension Subprocess {
+    public enum TerminationStatus: Sendable, Hashable {
+        #if canImport(WinSDK)
+        public typealias Code = DWORD
+        #else
+        public typealias Code = CInt
+        #endif
+
+        #if canImport(WinSDK)
+        case stillActive
+        #endif
+
+        case exit(Code)
+        case unhandledException(Code)
+
+        public var isSuccess: Bool {
+            switch self {
+            case .exit(let exitCode):
+                return exitCode == 0
+            case .unhandledException(_):
+                return false
+            }
+        }
+
+        public var isUnhandledException: Bool {
+            switch self {
+            case .exit(_):
+                return false
+            case .unhandledException(_):
+                return true
+            }
+        }
+    }
+}
+
+// MARK: - Internal
+extension Subprocess {
+    internal enum StringOrRawBytes: Sendable, Hashable {
+        case string(String)
+        case rawBytes([CChar])
+
+        // Return value needs to be deallocated manually by callee
+        func createRawBytes() -> UnsafeMutablePointer<CChar> {
```

Comment:
There are a decent number of functions here that return pointers allocated that the caller needs to manually deallocate. In Swift I think we normally prefer the closure-taking-function approach so that we can allocate, call the closure with the pointer, then deallocate and not rely on the caller to deallocate. This feels like a great use for that with something like a `withUnsafeBytes` function - do you think that'd work well here?